sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkBatchClient.cs

```diff
@@ -1,41 +1,243 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+// <auto-generated/>
+
+#nullable disable
+
 using System;
 using System.Threading;
 using System.Threading.Tasks;
+using Azure;
 using Azure.Analytics.Synapse.Spark.Models;
 using Azure.Core;
 using Azure.Core.Pipeline;
 
 namespace Azure.Analytics.Synapse.Spark
 {
-    [CodeGenSuppress("CreateSparkBatchJob", typeof(SparkBatchJobOptions), typeof(bool?), typeof(CancellationToken))]
-    [CodeGenSuppress("CreateSparkBatchJobAsync", typeof(SparkBatchJobOptions), typeof(bool?), typeof(CancellationToken))]
+    /// <summary> The SparkBatch service client. </summary>
     public partial class SparkBatchClient
     {
-        public virtual async Task<SparkBatchOperation> StartCreateSparkBatchJobAsync(SparkBatchJobOptions sparkBatchJobOptions, bool? detailed = null, CancellationToken cancellationToken = default)
-            => await StartCreateSparkBatchJobInternal (true, sparkBatchJobOptions, detailed, cancellationToken).ConfigureAwait(false);
+        private readonly ClientDiagnostics _clientDiagnostics;
+        private readonly HttpPipeline _pipeline;
+        internal SparkBatchRestClient RestClient { get; }
+
+        /// <summary> Initializes a new instance of SparkBatchClient for mocking. </summary>
+        protected SparkBatchClient()
+        {
+        }
+
+        /// <summary> Initializes a new instance of SparkBatchClient. </summary>
+        /// <param name="endpoint"> The workspace development endpoint, for example https://myworkspace.dev.azuresynapse.net. </param>
+        /// <param name="sparkPoolName"> Name of the spark pool. </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="livyApiVersion"> Valid api-version for the request. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        public SparkBatchClient(Uri endpoint, string sparkPoolName, TokenCredential credential, string livyApiVersion = "2019-11-01-preview", SparkClientOptions options = null)
```

Comment:
Also I am assuming the service version will need to fold into the SparkClientOptions now.

Reply:
yeah this is strange to see it here, not sure how the generator did this.